runtime/near-vm-errors/src/lib.rs

```diff
@@ -34,13 +57,9 @@ pub enum FunctionCallError {
     WasmUnknownError,
     HostError(HostError),
     EvmError(EvmError),
-    /// An error message when wasmer 1.0 returns a wasmer::RuntimeError
-    WasmerRuntimeError(String),
-    /// A trap in Wasmer 1.0, not same as WasmTrap above, String is a machine readable form like "stk_ovf"
-    /// String is used instead of wasmer internal enum is because of BorshSerializable.
-    /// It can be convert back by wasmer_vm::TrapCode::from_str
-    Wasmer1Trap(String),
+    ExecutionError(String),
```

Comment:
I'd do

```suggestion
    ExecutionError { msg: String },
```

just in case. This thing is serializable to JSON, and, tuple variants are not extensible:


```rust
#[derive(serde::Serialize, serde::Deserialize)]
enum E1 {
    V(String),
}

#[derive(serde::Serialize, serde::Deserialize)]
enum E2 {
    V { msg: String },
}

#[derive(serde::Serialize, serde::Deserialize)]
enum E3 {
    V { msg: String, code: u32 },
}


fn main() {
    println!("{}", serde_json::to_string(&E1::V("hello".into())).unwrap());
    println!("{}", serde_json::to_string(&E2::V { msg: "hello".into() }).unwrap());
    println!("{}", serde_json::to_string(&E3::V { msg: "hello".into(), code: 92 }).unwrap());
}

// OUTPUT
{"V":"hello"}
{"V":{"msg":"hello"}}
{"V":{"msg":"hello","code":92}}
```

Migrating `E2 -> E3` is easier that `E1 -> E3`.

Now, because this is also borsh serializable, this might not be super-relevant, as adding a field to a borsh struct is not compatible, but it's better to be on the safe side here (and consistent with `LinkError`).

Reply:
OK, done.